[lldb-dap] Test gardening, enabling tests and improving doc comments.

ashgti opened the pull request:
A bit of test gardenining. Enabling tests that were marked as skipped as part of https://github.com/llvm/llvm-project/issues/137660. Fixed up some comments and doc comments and fixed some test helpers.

da-viper commented:
@ashgti  I haven't check this yet because of #140788,  before remove the skipifs.

JDevlieghere commented on lldb/packages/Python/lldbsuite/test/decorators.py:
```diff
@@ -1102,3 +1103,28 @@ def is_feature_enabled():
     return skipTestIfFn(is_feature_enabled)
+
+
+def skipIfBinaryToLarge(path: Optional[str], maxSize: int):
+    """Skip the test if a binary is to large.
+
+    We skip this test for debug builds because it takes too long
+    parsing lldb's own debug info. Release builds are fine.
+    Checking the size of the lldb-dap binary seems to be a decent
+    proxy for a quick detection. It should be far less than 1 MB in
+    Release builds.
+    """
+
+    def check_binary_size():
+        if not path or not os.path.exists(path):
+            return "invalid path"
+
+        try:
+            size = os.path.getsize(path)
+            if size <= maxSize:
+                return None
+            return f"binary {path} (size = {size} is to larger than {maxSize}"
+        except:
+            return f"failed to read size of {path}"
+
+    return skipTestIfFn(check_binary_size)
```
This feels a little bit too ad-hoc. How about `@skipIfDebugBuild`? CMake knows if we're in a debug build and could configure lit, which in turn could tell detest. That seems like a more general solution to this particular problem.

da-viper replied:
[build_type.txt](https://github.com/user-attachments/files/20377561/build_type.txt)

I created a patch file with the basic build type. you can adjust it to match the correct build type.

ashgti replied:
I'll give this a shot and update this PR later today.

ashgti replied:
Done.

ashgti commented on lldb/test/API/tools/lldb-dap/console/TestDAP_console.py:
```diff
@@ -1,26 +1,23 @@
-
-    self.assertTrue(False, "No subprocess with name %s found" % process_name)
+class TestDAP_console(lldbdap_testcase.DAPTestCaseBase):
+    def get_subprocess(self, root_process, process_name):
```
This is using `self.assert` so, I don't think we want it as a staticmethod. I suspect that it was part of the class at one point because of the `self.assertTrue` that used to be on line 20.

ashgti commented:
This patch has grown a bit due to discovering a number of issues in some of the tests.

I think I'll try splitting it up into smaller patches instead.

ashgti committed:
[lldb-dap] Experimenting with async in DAP tests.

This is not fully tested yet, but the basic approach is to remove the background thread that can cause races in the tests.

Instead, I am using an asyncio event loop to read packets, and the tests can await on the loop to process packets as needed.

With further refactoring we should be able to push the async operations further up the stack into the test cases themselves.
